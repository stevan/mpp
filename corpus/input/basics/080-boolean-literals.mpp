# Boolean literals - true and false
my $is_valid = true;
my $is_done = false;

# Boolean in conditional
if ($is_valid == true) {
    print("valid");
}

if ($is_done != false) {
    print("not done");
}

# Boolean in expressions
my $and_result = true && false;
my $or_result = true || false;

# Boolean in ternary
my $check = $value eq "test" ? true : false;

# Boolean assignment
$flag = true;
$status = false;