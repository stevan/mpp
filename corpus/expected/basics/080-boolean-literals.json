[
  {
    "type": "Declaration",
    "declarator": "my",
    "variable": {
      "type": "Variable",
      "name": "$is_valid"
    },
    "initializer": {
      "type": "Boolean",
      "value": true
    }
  },
  {
    "type": "Declaration",
    "declarator": "my",
    "variable": {
      "type": "Variable",
      "name": "$is_done"
    },
    "initializer": {
      "type": "Boolean",
      "value": false
    }
  },
  {
    "type": "If",
    "condition": {
      "type": "BinaryOp",
      "operator": "==",
      "left": {
        "type": "Variable",
        "name": "$is_valid"
      },
      "right": {
        "type": "Boolean",
        "value": true
      }
    },
    "thenBlock": [
      {
        "type": "Call",
        "name": "print",
        "arguments": [
          {
            "type": "String",
            "value": "\"valid\""
          }
        ]
      }
    ],
    "elseIfClauses": []
  },
  {
    "type": "If",
    "condition": {
      "type": "BinaryOp",
      "operator": "!=",
      "left": {
        "type": "Variable",
        "name": "$is_done"
      },
      "right": {
        "type": "Boolean",
        "value": false
      }
    },
    "thenBlock": [
      {
        "type": "Call",
        "name": "print",
        "arguments": [
          {
            "type": "String",
            "value": "\"not done\""
          }
        ]
      }
    ],
    "elseIfClauses": []
  },
  {
    "type": "Declaration",
    "declarator": "my",
    "variable": {
      "type": "Variable",
      "name": "$and_result"
    },
    "initializer": {
      "type": "BinaryOp",
      "operator": "&&",
      "left": {
        "type": "Boolean",
        "value": true
      },
      "right": {
        "type": "Boolean",
        "value": false
      }
    }
  },
  {
    "type": "Declaration",
    "declarator": "my",
    "variable": {
      "type": "Variable",
      "name": "$or_result"
    },
    "initializer": {
      "type": "BinaryOp",
      "operator": "||",
      "left": {
        "type": "Boolean",
        "value": true
      },
      "right": {
        "type": "Boolean",
        "value": false
      }
    }
  },
  {
    "type": "Declaration",
    "declarator": "my",
    "variable": {
      "type": "Variable",
      "name": "$check"
    },
    "initializer": {
      "type": "Ternary",
      "condition": {
        "type": "BinaryOp",
        "operator": "eq",
        "left": {
          "type": "Variable",
          "name": "$value"
        },
        "right": {
          "type": "String",
          "value": "\"test\""
        }
      },
      "trueExpr": {
        "type": "Boolean",
        "value": true
      },
      "falseExpr": {
        "type": "Boolean",
        "value": false
      }
    }
  },
  {
    "type": "Assignment",
    "left": {
      "type": "Variable",
      "name": "$flag"
    },
    "operator": "=",
    "right": {
      "type": "Boolean",
      "value": true
    }
  },
  {
    "type": "Assignment",
    "left": {
      "type": "Variable",
      "name": "$status"
    },
    "operator": "=",
    "right": {
      "type": "Boolean",
      "value": false
    }
  }
]